public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect, useLayoutEffect } from 'react';
+
+import { getIndexPattern } from '../../overview/mitre/lib'
+import { Filter } from '../../../../../../src/plugins/data/public/';
+import { FilterMeta, FilterState, FilterStateStore } from '../../../../../../src/plugins/data/common';
+
+import {
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiSwitch,
+} from '@elastic/eui';
+
+//@ts-ignore
+import { getDataPlugin } from '../../../kibana-services';
+import { KbnSearchBar } from '../../kbn-search-bar';
+import { TimeRange, Query } from '../../../../../../src/plugins/data/common';
+import { Combobox } from './components'
+
+export const CustomSearchBar = ({ filtersValues, ...props }) => {
+
+    const KibanaServices = getDataPlugin().query;
+    const filterManager = KibanaServices.filterManager;
+    const timefilter = KibanaServices.timefilter.timefilter;
+    const indexPattern = getIndexPattern();
+    const [filterParams, setFilterParams] = useState({
+        filters: filterManager.getFilters().map(({ meta: { removable, ...restMeta }, ...rest }) => ({ ...rest, meta: restMeta })) || [],
+        query: { language: 'kuery', query: '' },
+        time: timefilter.getTime(),
+    });
```

Comment:
Why is this a single state object instead of using `useState` three times? I notice how later on the object is deconstructed to keep the values of the unchanged fields